stravalib/model.py

```diff
@@ -818,19 +1111,37 @@ class Activity(
     )
 
     @lazy_property
-    def comments(self):
+    def comments(self) -> BatchedResultsIterator[ActivityComment]:
+        assert self.bound_client is not None
         return self.bound_client.get_activity_comments(self.id)
 
+    # TODO: question - this returns a list of model.BaseActivityZone objects
```

Comment:
pls ignore this comment - i asked it above and can clean up this comment once it's answered above :) .  there are a few errors that pop up over and over in this module. once i better understand them i can fix throughout!